geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java

```diff
@@ -457,7 +471,7 @@ private void processOneConnection(Socket socket, long startTime, DataInputStream
 
       handler.endRequest(request, startTime);
 
-      startTime = DistributionStats.getStatTime();
+      startTime = nanoTimeSupplier.get();
```

Comment:
It's wrong to modify a method parameter (`startTime` in this case).

recommend making the method parameter `final` and defining a separate `final` variable on this line e.g. `startTime2`